app/services/file_attachment_blob_service.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class FileAttachmentBlobService
+  attr_reader :file, :revision, :replacement_file
+
+  def initialize(file:, revision:, replacement_file: false)
+    @file = file
+    @revision = revision
+    @replacement_file = replacement_file
+  end
+
+  def blob_id
+    blob.id
+  end
+
+  def blob_filename
```

Comment:
Oh I see this is probably called this because we have a private filename method. I'd just suggest having the one filename method and have it memo the filename, it doesn't need to call the filename from the blob object as I don't believe that does any transformation.

Reply:
This is renamed and extended in the next commit.